### ISSUE TITLE
Library search filter options

### ISSUE THREAD

**Issue by lisettemeij** [landportal/drupal#322]
Hi,

The filter options on this page: https://landportal.info/library/search are not always very intuitive. A couple of issues to keep in mind:

1. There seems to be a limit of maximum entries in the list, meaning that you cannot access all the countries or all the concepts for example. If I'm doing research on a particular country, it is not very intuitive (and I don't think directing people to the country page first is a solution, people should be able to find in 'search' what they are looking for and not have to look to other pages/sections).

2. The order of the inputs seems to be most resources on top. While for data providers, publishers and resource types this may make sense, I don't think it does for countries. You really have no idea where to search for the country that you are doing research on. So I would suggest to do that in alphabetical order (and/or make use of the regions hierarchy that is implemented in the back end + what you see when you add a resource to the Portal in the auto complete field). I am not sure what the most intuitive way to input 'concepts' here would be.... Happy to hear thoughts.

3. 'Data provider' is on top of the filters, while this is probably the item that means something to the fewest amount of people and will be least used. Can we move that down to the bottom?

4. There is currently no way to combine the search box and the filters on the bottom in one go. You now need to enter one first and then filter for the other. The most intuitive way would be to be able to select it all in one search entry and then click 'search' to go for your targeted and filtered search query.

5. It is also not possible to select one or more of the filters. Especially with elements like 'concepts' or 'countries', this seems like something you would want to include. Also, I would like to be able to filter out 'FAO legal office' as a data provider only for example, and then simply come up with the rest. We need to figure out a way to make these search options possible and the least frustrating to come up to the search we want :) A thing to consider here is that the property of several selections (meaning AND or OR) will be different for different filters. I can image for most it would be OR, but maybe for like concepts/OAs/Themes we would want 'AND'..?

6. Now when you have a search result, say: https://landportal.info/library/search/?f%5B0%5D=sm_field_doc_provider%3Anode%3A14232, if you enter something in the search bar above, it creates a new search and does not allow you to narrow down in the search that you are. We should probably enable that and at the same time allow for a 'Reset' or 'Clear' button to start from scratch. 

7. On the filter bar on the right, we should include year of publication as a filter (allows to add several, I would say OR behavior), publishers (OR behavior?), resource language (OR), data providers (OR). 

Some things to consider for the LL redesign and improvements.

**Comment by lisettemeij** [comment]
We should probably also consider the link between searching in the library alone, and the overall website search. Often that is where I start as it's on the top right of every screen.

Then when I filter 'LL resource', I immediately want to get into the filters that I mentioned above. But for that, you need to go to an entirely different page.

Probably we should think about whether we should (and if so, how) we can incorporate the two searches together OR (seemingly easier solution) to direct users to the Library search page once they've selected this filter.

**Comment by jcr** [comment]
For this release, from this ticket a few , possible things, have been addressed
 - filters / facets order have been adjusted 
 - added Licenses and Language
3, 4 & 6 should also be addressed (now hove a 'keep filter' option box under the search bar + providers are hidden as a side filter)
Now there are way too many different things and issues on this ticket so I opened some new ones:
For date filter / #408
For UI (point 2.): #409
For filters/facet behavior (point 1, 5 & 7): #410

For the last comment you put this is to be considered. It is possible on the 'main' search (I think) to have 'dynamic' that appears as you select specific resources / option combination. For instance, once you select a LLR , news or something that have a 'publication date' we could show automatically some more specific filters.
At one point we'll need to have some real in depth look at our search interface, filter,  behavior, UI and so on. (project for 2018?)

closing this one ticket for now ;)
